### PR TITLE
Func test util image format

### PR DESCRIPTION
**Pull request by Arc-Celt**
- Add conditions that check if input is a NumPy array in `_input_checker`. Raises TypeError if not.
- Add unit test cases for the image format checks.2 expected cases, 2 edge cases, 3 erroneous cases.

P.S. I found out why pytest isn't working. It has something to do with the `__init__.py` under `src\sharpedge`:
```
__version__ = version("sharpedge")
```
Daniel provided the correct approach today. That is, before `pytest`, run `poetry install` first so that the version could be correctly acquired.

### CODE REVIEW

File: src/sharpedge/_utils/utility.py [hunk 1]
```diff
@@ -1,19 +1,13 @@
...
-            return False
-        elif "Archer": # Update the condtion
-            # Rasie error message
-            return False
-        elif "Inder": # Update the condtion
-            return False
+        # Check the image format: must be a numpy array
+        if not isinstance(img_array, np.ndarray):
+            raise TypeError("Image format must be a numpy array.")
         return True
```
Comment (JennyonOort) [inline src/sharpedge/_utils/utility.py:13]: Consider deleting as this would be redundant output to users.

File: tests/test_utility.py
Comment (JennyonOort) [inline tests/test_utility.py:1]: Great job on following the AAA practice! A couple thoughts for you to consider:

1) take advantage of `pytest` functions such as `pytest.mark.parametrize`, `pytest.raise`, and `pytest.fail`. These could significantly reduce duplication of testing code
2) try consolidating functions by valid, edge, and error cases, so instead of having a function for each test, each type of test is bundled to one
3) For each of the test, would have at least two test data example (to echo what Tiff said about how the func may just be working for one dataset). one grayscale and one rgb should do. More are welcomed of course. x

On the test cases, I have the following additions to recommend:
- edge case: np.array with non-standard dtype (for ex. float) for rgb and grayscale => should pass your check and ultimately rejected in the color range check
- edge case: empty image for rgb and grayscale => should pass your check but rejected in the shape non-zero test
- error case: test with tuples
- error cases: test with `None` (`NoneType` object)

**Review by JennyonOort — CHANGES_REQUESTED** [review]
A few thoughts from my side. Happy to discuss whereever needs more clarity!

File: src/sharpedge/_utils/utility.py (hunk 1 above)
Comment (Arc-Celt) [reply src/sharpedge/_utils/utility.py:13]: > Consider deleting as this would be redundant output to users.

Because I don't have a separate function, I think this part should be kept in the `_input_checker` method?  I saw that Inder's implementation also follows this approach, and I think it wouldn't lead to redundancy. Feel free to discuss more about this later today!

File: tests/test_utility.py
Comment (Arc-Celt) [reply tests/test_utility.py:1]: Thanks for the great advice on using the `pytest.parametrize decorator` to test with all parameter combinations at once! I've also included the test cases you suggested. Feel free to review them!

**Review by hankunxiao — APPROVED** [review]

**Review by JennyonOort — APPROVED** [review]
Looks good to me! 

As just discussed over our Zoom Meeting on updated _input_checker function structure, please make sure to remove `return True` before or after merging.
